### PR TITLE
[Merged by Bors] - chore(SobolevInequality): drop a `DecidableEq` assumption

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/FunctionalSpaces/SobolevInequality.lean [hunk 1]
```diff
@@ -294,6 +294,7 @@ theorem lintegral_pow_le_pow_lintegral_fderiv_aux
     {u : (ι → ℝ) → F} (hu : ContDiff ℝ 1 u)
     (h2u : HasCompactSupport u) :
     ∫⁻ x, (‖u x‖₊ : ℝ≥0∞) ^ p ≤ (∫⁻ x, ‖fderiv ℝ u x‖₊) ^ p := by
+  letI := Classical.decEq ι
```
Comment (grunweg) [inline Mathlib/Analysis/FunctionalSpaces/SobolevInequality.lean:297]: Would you like to also drop the `DecidableEq ι` assumption on line 57?

**Review by grunweg — COMMENTED** [review]
One quick question - otherwise, looks good to me. Thanks!

File: Mathlib/Analysis/FunctionalSpaces/SobolevInequality.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Analysis/FunctionalSpaces/SobolevInequality.lean:297]: The definition of `T` and some others need decidable equality.

Comment (j-loreaux) [reply Mathlib/Analysis/FunctionalSpaces/SobolevInequality.lean:297]: I guess this is taking advantage of the current behavior of `variable` which drops unnecessary variables after processing the declaration. Can you just shadow the variables instead?

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**